lib/Client.php

```diff
@@ -127,9 +128,46 @@ public function getUserInfo() {
 			return $this->getAccessTokenPayload();
 		}
 
+		if (isset($openIdConfig['use-access-token-introspection-for-user-info']) && $openIdConfig['use-access-token-introspection-for-user-info']) {
+			$introspectionClientId = $openIdConfig['token-introspection-endpoint-client-id'] ?? null;
+			$introspectionClientSecret = $openIdConfig['token-introspection-endpoint-client-secret'] ?? null;
+			$accessToken = $this->getAccessToken();
+			if (isset($openIdConfig['exchange-token-mode-before-introspection'])) {
+				$mode = $openIdConfig['exchange-token-mode-before-introspection'];
+				$token = $mode === 'refresh-token' ? $this->getRefreshToken() : $this->getAccessToken();
```

Comment:
handle case where we have no refresh token -> throw exception/log error what ever.

This happens if oc is a rely party